go-controller/pkg/util/kube.go

```diff
@@ -220,6 +250,72 @@ func newKubernetesRestConfig(conf *config.KubernetesConfig) (*rest.Config, error
 	return kconfig, nil
 }
 
+// startNodeCertificateManager manages the creation and rotation of the node-specific client certificate.
+// When there is no existing certificate, it will use the BootstrapKubeconfig kubeconfig to create a CSR and it will
```

Comment:
Is this `When there is no existing certificate` handled by the certManager?

Let's say someone places valid certs there to start with and passes the appropriate option along with the bootstrap option, what would happen? Out of curiosity but also not sure if we should do something in this case.

Reply:
The certs would be used, this is the desired behavior. When the pod restarts we want it to use the existing certs if they are still valid instead of always requesting new ones.